net-mail/cyrus-imapd/cyrus-imapd-3.12.0.ebuild

```diff
@@ -0,0 +1,271 @@
+REQUIRED_USE="
+	afs? ( kerberos )
+	calalarm? ( http )
+	http? ( xapian )
+"
+
+COMMON_DEPEND="
+	dev-libs/libpcre2:=
+	>=dev-libs/cyrus-sasl-2.1.13:2
+	>=dev-libs/jansson-2.10:=
+	>=dev-libs/icu-55:=
+	dev-libs/libical:=
+	sys-apps/util-linux
+	sys-fs/e2fsprogs
+	sys-libs/zlib:=
+	afs? ( net-fs/openafs )
+	caps? ( sys-libs/libcap )
+	clamav? ( app-antivirus/clamav:= )
+	http? (
+		app-arch/brotli:=
+		app-arch/zstd:=
+		dev-libs/libxml2
+		>=net-libs/nghttp2-1.34.0:=
+		sci-libs/shapelib:=
+	)
+	kerberos? ( virtual/krb5 )
+	ldap? ( net-nds/openldap:= )
+	mysql? ( dev-db/mysql-connector-c:= )
+	nntp? ( !net-nntp/leafnode )
+	pam? (
+		>=net-mail/mailbase-1
+		sys-libs/pam
+	)
+	perl? ( dev-lang/perl:= )
+	postgres? ( dev-db/postgresql:* )
+	ssl? ( >=dev-libs/openssl-1.0.1e:=[-bindist(-)] )
+	dev-db/sqlite:3
+	tcpd? (
+		net-libs/libnsl:=
+		>=sys-apps/tcp-wrappers-7.6
+	)
+	xapian? (
+		>=dev-libs/xapian-1.4.0:=
+		net-misc/rsync
+	)
+"
+RDEPEND="${COMMON_DEPEND}
+	acct-group/mail
+	acct-user/cyrus
+"
+DEPEND="${COMMON_DEPEND}
+	test? ( dev-util/cunit )
+"
+# app-editors/vim-core needed for xxd
+BDEPEND="
+	app-alternatives/lex
+	app-alternatives/yacc
+	virtual/pkgconfig
+	http? ( app-editors/vim-core )
+"
+
+# https://bugs.gentoo.org/678754
+# TODO: check underlinking for other libraries
+#PATCHES=(
+#	"${FILESDIR}/cyrus-imapd-libcap-libs-r1.patch"
+#)
+
+PATCHES=(
+	"${FILESDIR}"/cyrus-imapd-3.6.7-O3-perl.patch
+	"${FILESDIR}"/cyrus-imapd-3.12.0-c21.patch
+)
+
+# Follow upstream recommendations with binary naming
+# https://www.cyrusimap.org/imap/download/packagers.html
+declare -A RENAME_BINARY=(
+	[ptdump]="usr/sbin;8" # bug #905765
+	[master]="usr/libexec;8"
+)
+
+src_prepare() {
+	default
+
+	local bindir mansection
+	for binary in "${!RENAME_BINARY[@]}"; do
+		IFS=';' read -r bindir mansection <<< "${RENAME_BINARY[${binary}]}"
+		for ${file} in $(grep -rl -e "${binary}(${mansection})" man); do
+			sed -e "s/${binary}(${mansection})/cyr_${binary}(${manpage})/" \
+				-i ${file} || die
+		done
+		sed -e "s/${binary}.${mansection}/cyr_${binary}.${mansection}/" \
+			-i Makefile.am || die
+		sed -e "/.SH NAME/,/.sp/ s/^${binary}/cyr_${binary}/" \
+			-e "s/\\\fB${binary}\\\fP/\\\fBcyr_${binary}\\\fP/" \
+			-e "s/${binary^^}/CYR_${binary^^}/" \
+			-i man/${binary}.${mansection} || die
+		mv man/${binary}.${mansection} man/cyr_${binary}.${mansection} || die
+	done
+
+	# lock.h to afs/lock.h
+	sed -i -e '/lock.h/s:lock.h:afs/lock.h:' \
+		ptclient/afskrb.c || die
+
+	eautoreconf
+}
+
+src_configure() {
+	# bug #604466
+	append-ldflags $(no-as-needed)
+
+	# Workaround runtime crash
+	# bug #834573
+	append-flags -fno-toplevel-reorder
+
+	# lto-type-mismatch
+	filter-lto
+
+	# TODO:
+	# - revisit --with-sphinx-build=no? (it's docs this time, not the search engine)
+	# - post-emerge message re lmdb removal?
+	local myconf=(
+		--enable-murder
+		--disable-release-checks
+		--enable-idled
+		--enable-autocreate
+		--enable-pcre
+		--enable-pcre2
```

Comment:
shouldn't this be:

--disable-pcre \
--enable-pcre2 \

to avoid any ambiguity about which pcre we are building / linking against.